db/migrate/20190806102256_add_applications_open_from_to_courses.rb

```diff
@@ -0,0 +1,16 @@
+class AddApplicationsOpenFromToCourses < ActiveRecord::Migration[5.2]
+  def change
+    add_column :course, :applications_open_from, :date
+
+    say_with_time 'Setting applications_open_from to one held within sites' do
+      Course.all.each do |course|
+        if course.recruitment_cycle.current?
+          applications_open_from = course.site_statuses.order("applications_accepted_from ASC").first.applications_accepted_from
```

Comment:
shouldn't it be the whole of this?

```
  site_statuses
      .open_for_applications
      .order("applications_accepted_from ASC")
      .first
      &.applications_accepted_from
```

this gives the site status that is open for application as it is findable and application being accepted now with vacancies

Reply:
nvm
https://github.com/DFE-Digital/manage-courses-backend/pull/684#discussion_r311033152

Reply:
Theo and I chatted about it but we made the assumption that the check as to whether or not they're open isn't necessary, though this was back when we were creating a new field for it. 

Given we're now replacing the field - it would probably be worth keeping the current behaviour, i'll update this :tada:

Reply:
https://trello.com/c/l0t1Oun3/1699-applications-open-from-shouldnt-be-restricted-to-just-sites-that-are-open-to-applications